backend/apps/chatbot/urls.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+from django.urls import path
+
+from .views import (
+    ChatbotAskView,
+    ChatInteractionSaveView,
+    ClearAssistantMemoryView,
+    FeedbackSaveView,
+    FeedbackUpdateView,
+)
+
+urlpatterns = [
+    path(
+        "chatbot/ask",
+        ChatbotAskView.as_view(),
+        name="chatbot_ask",
+    ),
+    path(
+        "chatbot/interactions/save",
+        ChatInteractionSaveView.as_view(),
+        name="save_chat_interaction",
+    ),
+    path(
+        "chatbot/feedback/save",
```

Comment:
Change to:
PUT chatbot/threads/{thread_id}/feedback

PUT is idempotent, so it does both save and update